### DISCUSSION TITLE
Playground notebooks

### DISCUSSION THREAD

**Discussion by phil65** [facelessuser/pymdown-extensions#2142] [Q&A]
Hello!
First of all, thanks for all the awesome stuff you´re doing.
I have a question: Is there an easy way to integrate these playground notebooks in other projects?
I tried ripping it out of ColorAide by copying all the javascript/css stuff and removing the color-related stuff from notebook.py,
and I got it to show up and load, but I didnt get the buttons ("edit" etc) below the code block working.
Thank you!

**Comment by facelessuser** [comment]
So, I've converted this to a discussion as it isn't really related to a feature request or bug.

It sounds like you just want a generic Python console (no color stuff)? If so, the code can be dramatically simplified (removing loop processing and a number of things). I only had to process certain structures to get at the color objects. Remove those, and this all becomes much more simple.

As for why the buttons do not show up, I cannot say as I have no idea what you've "exactly" done. I currently do not document how I do what I do, and while anyone is free to dig in and attempt to use it, I have no guides on how to reuse it and do not currently offer support for people to use it. I have been working on a non-color version, but it is not public at this time. Ideally, something like this would be integrated as a documentation plugin for public consumption, but currently, it is very specialized for this specific project.

**Comment by phil65** [comment]
Okay, thank you for the info.
To give a bit more context:
I wrote a MkDocs plugin to build Websites in an object-oriented way (it´s still WIP, but you can see the current state here: https://phil65.github.io/mknodes/ ).
That plugin also leverages a lot of the Markdown Extensions you wrote, so thank you for that. :)
Ideally I would want to integrate a playground into each of the node pages, so that the user can check out each node.
Instead of rendering a color, it would render the given Markdown then.
(To help a bit to understand what I mean, take this as an example:
 https://phil65.github.io/mknodes/The%20nodes/Presentation%20nodes/MkDiagram.html )

I will follow your work then and be the first if you will release anything in that direction 😄

**Comment by phil65** [comment]
Oh, and one more question:
I also wrote nodes for your new Block Extensions, but for Material, I cant get the tab block extension to work.
(See here: https://phil65.github.io/mknodes/The%20nodes/Block%20nodes/MkTabbedBlocks.html )
Is it correct that it is currently not supported?

**Reply by facelessuser** [reply]
Can you be a bit more clear with the question? You are saying this extension (https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/tab/) does not work? If so, it does work as I use it all the time. It is possible that I misunderstand what you mean by "does not work". Any context you can provide will be helpful.

**Reply by phil65** [reply]
Correct, I mean that extension.

I added the block extension to mkdocs.yml,
the generated markdown looks like:
```
/// tab | Tab 1
    new: True

Some markdown
///

/// tab | Tab 2

Other Markdown
///
```

But it doesnt render as Tabs.

You can compare the "Markdown" and "Rendered" tabs of these two pages:
https://phil65.github.io/mknodes/The%20nodes/Block%20nodes/MkTabbedBlocks.html (not working)
https://phil65.github.io/mknodes/The%20nodes/Block%20nodes/MkAdmonitionBlock.html (working)

**Reply by facelessuser** [reply]
Where is the raw source. I need to look at the Markdown with context of other stuff in the page.

**Reply by phil65** [reply]
There is no raw source, all the markdown for the whole website is generated by code. 😄
But you can see the generated markdown for each page at the bottom of each page, in a collapsed details box.
(ignore the outer `/// html | div` block)

**Reply by facelessuser** [reply]
1. I would not use `tabbed` and `blocks.tab` together.
2. I need raw source, with context, if you want me to debug.

Run either `tabbed` or `blocks.tab`, not both, and please provide full source and context.

**Reply by facelessuser** [reply]
Also, this is really off-topic for ColorAide. Please move this conversation over ot Pymdown Extensions.

**Reply by facelessuser** [reply]
Okay, I sent this over to Pymdown Extensions. One other thing I can think of is maybe you are not nesting blocks proper? If one block is nested under another, you need to have a different number `///`.

```
/// tab | one
//// tab | nested

content
////
///
```

**Reply by phil65** [reply]
I wouldnt use them together usually...It´s just for the documentation. :)

Yes, my code takes nesting into account exactly in the way you describe.
( https://github.com/phil65/mknodes/blob/5a05adccd921907ec7488962d50bd0467981971d/mknodes/basenodes/mkblock.py#L43C17-L43C17 )
I pushed a quick static page with a single tab block. 
https://phil65.github.io/mknodes/The%20nodes/MkNav/From%20file/test_file.html
https://raw.githubusercontent.com/phil65/mknodes/5a05adccd921907ec7488962d50bd0467981971d/tests/data/nav_tree/test_file.md

What I find confusing is that it works with the "details" and "admonition" block extensions, just not with tabs..

**Reply by facelessuser** [reply]
Unfortunately, you aren't providing me with enough to debug. Try a minimal example, stripping out unnecessary plugins. I can't debug your dynamic environment.

**Reply by facelessuser** [reply]
Anyways, as far as I can see, nested tabs work fine. Here's a live example:

https://facelessuser.github.io/coloraide/playground/?notebook=https%3A%2F%2Fgist.githubusercontent.com%2Ffacelessuser%2F837b79965348d183bbcc72b28aecf7d9%2Fraw%2F2c1c57a20a51c22b985eb198cfe350ae8a5f3e38%2Ftabs.md#

Click the Markdown button to see the source.

**Reply by facelessuser** [reply]
Sigh, there is a bug apparently when using the Markdown content in playground page where you can't edit the source (or view it). Anyways, here is the source:

```
/// tab | Tab 1

//// tab | Tab A
content
////

//// tab | Tab B
content
////
///

/// tab | Tab 2

Other conent
///
```

**Reply by phil65** [reply]
I pushed this: https://github.com/phil65/mknodes/commit/07185d012cf12a5babdae6c0439ca37a8d2df51b
And it ends up with: https://phil65.github.io/mknodes/test_file.html
Even when locally removing all plugins (including my own one), markdown extensions, extra css/whatever, tab blocks dont work for me.

**Reply by facelessuser** [reply]
Again, I need a reproducible example, what can I run to reproduce? I don't even know what your configuration is. I can assure you it works. I have tests verifying this and I just liked you to a working example. This is most likely a user error, but I don't know what you are doing as you have not provided me a reproducible example.

**Reply by phil65** [reply]
Could you give me a hint what you would describe as a reproducible example?
I could push a basically empty mkdocs.yml if you like. 😄

**Reply by facelessuser** [reply]
Give me a zip with a simple mkdocs project, stripped down mkdocs.yml and a single document page with your issue.

**Reply by phil65** [reply]
Found the solution by checking your code.
```
    alternate_style: true
```
is required, for whatever reason.

**Reply by facelessuser** [reply]
Material has requested an alternate output for styling. The legacy tabs had then, and so do the new block ones for backwards compatibility.

**Reply by phil65** [reply]
Should the "non-alternate" style also work for Material?

**Reply by facelessuser** [reply]
Not anymore, I don't think. I think it requires "alternate" now. It allows for scrollable tabs.

**Reply by phil65** [reply]
Ok, understood. Perhaps worth updating the documentation, that setting is described as "experimental" there. 😄
Thank you for your time!
